src/Simplify_Add.cpp

```diff
@@ -81,7 +81,7 @@ Expr Simplify::visit(const Add *op, ConstBounds *bounds) {
                rewrite((x + c0)/c1 + c2, (x + fold(c0 + c1*c2))/c1) ||
                rewrite((x + (y + c0)/c1) + c2, x + (y + (c0 + c1*c2))/c1) ||
                rewrite(((y + c0)/c1 + x) + c2, x + (y + (c0 + c1*c2))/c1) ||
-               rewrite((c0 - x)/c1 + c2, (fold(c0 + c1*c2) - x)/c1, c0 != 0) ||
```

Comment:
@jn80842 turns out the c0 != 0 was here to prevent this entering a loop with a different rule.